### ISSUE TITLE
Dice rolling automatically on render

### ISSUE THREAD

**Issue by timB73** [AdamTyler/react-dice-complete#23]
This seems to have cropped up from nowhere. It used to be that it wouldn't roll on render, but suddenly everywhere I try it the dice is automatically rolling with the value 4.

I was able to reproduce it in Sandbox, but it's weird because I'm not getting the same behaviour on my main production branch, using the same version and I can't figure out why.

https://codesandbox.io/s/busy-keldysh-oefwf?file=/src/App.js

**Comment by timB73** [comment]
Ok version 1.2.0 works perfectly, I guess I must have had different versions after all (I had updated my dev version to latest version after I encountered the problem to see if it would fix it, so not sure which version it was originally).

I looked through the code a bit, and in version 1.2.1 you introduce the defaultRoll prop, but you've set the default prop value of this to 4, so that's why I was getting 4. However I'm not sure the reason for the roll animation starting automatically on render.

**Comment by The1Percent** [comment]
Default roll is likely because you have are not calling it via a function like onClick={() => blah}. The two parenthesis are important.

**Comment by joey-carlisle4** [comment]
The dice are still rolling every time they re-render.  How can I prevent that?

**Comment by AdamTyler** [comment]
@timB73 can you retest with the latest version of the library and see if you are still seeing the issue?

**Comment by timB73** [comment]
Yes it still happens for me on the latest version. Tried changing the rollDone callback to parenthesis as @The1Percent suggested but didn't help. Also tested on a blank project and get the same.

**Comment by AdamTyler** [comment]
@timB73 I see what you are saying...

Whats happening is the library is building the die and then applying the rotation animation to make sure the initial value is facing up.

I will try and think of a way around this

**Comment by timB73** [comment]
Hello, just wondering if there are any updates on this? It would be great if this could get fixed.

**Comment by The1Percent** [comment]
Hey Tim,

I can’t write code outside of my current role without conflict of interest review and approval. Maybe one of the other team members can assist.

 

Thanks,

 

Shane

**Comment by badidu09** [comment]
Hello, just to know if there is update ? I have the same issue

**Comment by AdamTyler** [comment]
This issue has been fixed in v2.2.0

**Comment by timB73** [comment]
Awesome thank you!

**Comment by timB73** [comment]
Hi, btw I still seem to be getting the same behaviour after upgrading to v2.2.0
